include/matx/transforms/reduce.h

```diff
@@ -1229,52 +1229,77 @@ void __MATX_INLINE__ median(OutType dest,
+
+    MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
 
-  // If the rank is 0 we're finding the median of a vector
-  if constexpr (RANK_IN == 1) {
-    matx::sort(tmp_sort, in, SORT_DIR_ASC, stream);
+      auto tmp_sort = make_tensor<T>(in.Shape(), MATX_ASYNC_DEVICE_MEMORY, stream);
 
-    // Store median
-    if (tmp_sort.Lsize() & 1) {
-      auto middlev =
+    // If the rank is 0 we're finding the median of a vector
+    if constexpr (RANK_IN == 1) {
+      matx::sort(tmp_sort, in, SORT_DIR_ASC, stream);
+
+      // Store median
+      if (tmp_sort.Lsize() & 1) {
+        auto middlev =
           tmp_sort.template Slice<0>({tmp_sort.Lsize() / 2}, {matxDropDim});
-      matx::copy(dest, middlev, stream);
-    }
-    else {
-      auto middle1v =
+        matx::copy(dest, middlev, stream);
+      }
+      else {
+        auto middle1v =
           tmp_sort.template Slice<0>({tmp_sort.Lsize() / 2 - 1}, {matxDropDim});
-      auto middle2v =
+        auto middle2v =
           tmp_sort.template Slice<0>({tmp_sort.Lsize() / 2}, {matxDropDim});
-      (dest = (middle1v + middle2v) / 2.0f).run(stream);
+        (dest = (middle1v + middle2v) / 2.0f).run(stream);
+      }
     }
-  }
-  else if constexpr (RANK_IN == 2) {
-    MATX_ASSERT(dest.Size(0) == in.Size(0), matxInvalidSize);
+    else if constexpr (RANK_IN == 2) {
+      MATX_ASSERT(dest.Size(0) == in.Size(0), matxInvalidSize);
 
-    matx::sort(tmp_sort, in, SORT_DIR_ASC, stream);
+      matx::sort(tmp_sort, in, SORT_DIR_ASC, stream);
 
-    if (tmp_sort.Lsize() & 1) {
-      auto sv = tmp_sort.template Slice<1>({0, tmp_sort.Lsize() / 2},
-                                           {matxEnd, matxDropDim});
-      (dest = self(sv)).run(stream);
+      if (tmp_sort.Lsize() & 1) {
+        auto sv = tmp_sort.template Slice<1>({0, tmp_sort.Lsize() / 2},
+            {matxEnd, matxDropDim});
+        (dest = self(sv)).run(stream);
+      }
+      else {
+        auto sv = tmp_sort.template Slice<1>({0, tmp_sort.Lsize() / 2 - 1},
+            {matxEnd, matxDropDim});
+        auto sv2 = tmp_sort.template Slice<1>({0, tmp_sort.Lsize() / 2},
+            {matxEnd, matxDropDim});
+        (dest = (sv + sv2) / 2.0f).run(stream);
+      }
     }
-    else {
-      auto sv = tmp_sort.template Slice<1>({0, tmp_sort.Lsize() / 2 - 1},
-                                           {matxEnd, matxDropDim});
-      auto sv2 = tmp_sort.template Slice<1>({0, tmp_sort.Lsize() / 2},
-                                            {matxEnd, matxDropDim});
-      (dest = (sv + sv2) / 2.0f).run(stream);
+  } else {
+
+    constexpr int batch_dims = OutType::Rank() - 1;
+    constexpr int red_dims = OutType::Rank() - TensorInType::Rank();
+
+    if constexpr (batch_dims > 1) {
+      // collapse batch dimensions to a single dimension
+      auto oop = lcollapse<batch_dims>(dest);
+      auto iop = lcollapse<batch_dims>(in);
+
+      static_assert(oop.Rank() <= 1);
```

Comment:
Maybe add an error string on all these static asserts to tell them what's wrong?

Reply:
these are actually just sanity check asserts that should be impossible to kick up.  They are there for people who are modifying this specific code (aka validating my logic was correct).